### ISSUE TITLE
Axis labels starts at -0

### ISSUE THREAD

**Issue by paul1956** [kirsan31/winforms-datavisualization#33]
The chart only has positive y values though some are very small 0.0000000001 and I do absolute on all the data values (there is a lot of math to get them including subtraction, I only care about 2.25 digits of precision ie.. X.0, X.025,X.5 and X.075) but I can't get rid of the - sign before the 0. Is there a way I should be setting the YAxis labels to prevent this? How is the YAxis starting value set/calculated?

**Comment by kirsan31** [comment]
Hi @paul1956 
Yea it's a known _problem_ with axes auto min \ max values due to floating point calculation. In your situation I think you can simply set axis `Minimum` to 0 in designer, or in code once.
More of it, I suggest not to use Auto min \ max values if you really not need them. Because their calculation is quite a costly operation.

**Comment by paul1956** [comment]
Thanks. I don't need auto I just didn't know where to set a fixed value. I still create the graphs at runtime. Unless you have burning desire to fix this can be closed.

**Comment by paul1956** [comment]
@kirsan31 I think this is a bug, Below is chart initialization and AxixY.Minimum is set to 0
```
       With treatmentMarkersChartArea.AxisY
           Dim interval As Single = (TreatmentInsulinRow / 10).RoundSingle(3)
           .Interval = interval
           .IsInterlaced = False
           .IsMarginVisible = False
           .IsStartedFromZero = False
           With .LabelStyle
               .Font = labelFont
               .ForeColor = labelColor
               .Format = "{0.00}"
           End With
           .LineColor = Color.FromArgb(64, labelColor)
           With .MajorTickMark
               .Enabled = True
               .Interval = interval
               .LineColor = Color.FromArgb(64, labelColor)
           End With
           .Maximum = TreatmentInsulinRow
           .Minimum = 0
           .Title = "Delivered Insulin"
           .TitleFont = New Font(labelFont.FontFamily, 14)
           .TitleForeColor = labelColor
       End With
```

**Comment by kirsan31** [comment]
It's very strange...
About [`StartFromZero`](http://msdn.microsoft.com/en-us/library/system.windows.forms.datavisualization.charting.axis.isstartedfromzero.aspx):

> Gets or sets a flag which indicates whether the minimum value of the axis will be automatically set to zero if all data point values are positive. If there are negative data point values, the minimum value of the data points will be used.

Can you provide some repro project to I can debug this?

**Comment by paul1956** [comment]
The only project I have using the chart is my CareLink project, it does have sample data.
https://github.com/paul1956/CareLink
It has sample data under Start Here/Use Test  Data
and the 2 tabs Running IOB and Treatment Details have the charts showing -0

I don't program in C# well enough to create a repo

If its too much work just forget it.  The initialization code for the charts is very simple and I tried doing absolute value on all the Y values but it doesn't seem to matter.

**Comment by paul1956** [comment]
![image](https://user-images.githubusercontent.com/12789369/225600958-4f26ee87-8815-4925-980c-f5f42c68a910.png)
Here is what it looks like the Y axis plots the green line with black dots and the smallest value in the test data is 0.168 if the line every hits 0 the person using the pump is going to be needing medical attention so its extremely unlikely and the reason this chart exists and negative values are impossible.

**Comment by kirsan31** [comment]
Okay, as soon as I have time, I'll try to figure it out...

**Comment by kirsan31** [comment]
I was able to reproduce the problem, we'll see what can be done...

**Comment by kirsan31** [comment]
The cause:
![image](https://user-images.githubusercontent.com/17767561/226112885-fcdb8b87-2922-475f-9870-60653e26dc7d.png)
I think it's related to https://devblogs.microsoft.com/dotnet/floating-point-parsing-and-formatting-improvements-in-net-core-3-0/

Will be fixed in 1.8.

**Comment by paul1956** [comment]
@kirsan31 Thanks, it's perfect. I did try setting Axis.Minimum to Double.Epsilon before the fix but that didn't work.
